torch/csrc/jit/codegen/cuda/lower_index_hoist.cpp

```diff
@@ -12,419 +12,275 @@ namespace cuda {
 
 namespace {
 
-// Return leaf domains of a given domain.
-std::unordered_set<IterDomain*> getUsedLeafIds(
-    IterDomain* id,
-    TensorDomain* td) {
-  const auto all_vals_between = DependencyCheck::getAllValsBetween(
-      {id}, {td->domain().begin(), td->domain().end()});
-
-  std::unordered_set<IterDomain*> used_leaf_ids;
-
-  for (const auto leaf : td->domain()) {
-    if (std::find(all_vals_between.begin(), all_vals_between.end(), leaf) !=
-        all_vals_between.end()) {
-      used_leaf_ids.insert(leaf);
+// Given a set of loop variables (`dependency`), find the outer-most loop nest
+// that contains all these variables.
+int64_t findLowestLevelWithSatisfiedDependency(
```

Comment:
nit: I'd prefer to use common vocabulary, e.g., level -> position, lowest -> outermost

Reply:
fixed